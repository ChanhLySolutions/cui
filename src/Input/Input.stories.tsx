import { Meta } from '@storybook/react';
import { Input } from '.';
import { InputProps } from './Input';

const meta: Meta = {
  title: 'Components/Input',
  component: Input,
  parameters: {
    jest: ['Input.test.tsx'],
  },
};

export default meta;

export function Default(props: InputProps) {
  const { ...rest } = props;
  return <Input label='Default' {...rest} />;
}
