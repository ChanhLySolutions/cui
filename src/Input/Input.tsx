import { ThemeContext } from '@/contexts/ThemeContext';
import classNames from 'classnames';
import { useContext } from 'react';
import { Path, UseFormRegister } from 'react-hook-form';
import { Label } from './Label';

export interface InputProps<IFormValues = void> {
  name?: Path<IFormValues>;
  label?: string;
  placeholder?: string;
  register?: UseFormRegister<IFormValues>;
  size?: 'small' | 'medium' | 'large';
  required?: boolean;
}

export function Input<IFormValues = void>(props: InputProps<IFormValues>) {
  const { theme } = useContext(ThemeContext);
  const { name, label, placeholder, register, size = theme.size, required } = props;

  const className = classNames('border', {
    'px-1 py-0.5': size === 'small',
    'px-2 py-1': size === 'medium',
    'px-3 py-1.5': size === 'large',
  });

  return (
    <>
      <Label name={name}>{label}</Label>
      <input
        className={className}
        placeholder={placeholder}
        {...register?.(name!, { required })}
        required
        type='text'
      />
    </>
  );
}
