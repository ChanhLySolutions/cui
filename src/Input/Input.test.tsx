import renderer from 'react-test-renderer';
import { Input } from './Input';

it('renders correctly', () => {
  const input = renderer.create(<Input />).toJSON();
  expect(input).toMatchSnapshot();
});
