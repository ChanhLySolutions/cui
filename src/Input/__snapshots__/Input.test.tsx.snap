// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<input
  className="border px-2 py-1"
  required={true}
  type="text"
/>
`;
